models/tasks.js

```diff
@@ -284,6 +284,17 @@ const overdueTasks = async (overDueTasks) => {
     throw err;
   }
 };
+
+const generateTaskNumber = async () => {
+  try {
+    const snapshot = await tasksModel.count().get();
+    return snapshot.data().count + 1;
```

Comment:
How does this avoid race conditions?

Reply:
It doesn't handle race conditions. As of now isn't just the super user creating tasks using the dashboard site? For now, it won't cause race conditions, but in the future, if we are allowing many members or AOs to create then it might create race conditions. Also, you said till then we might migrate to better infrastructure. So, how do you want me to handle this?